details/operator-sdk/controllers/astraconnector_controller.go

```diff
@@ -195,19 +195,18 @@ func (r *AstraConnectorController) Reconcile(ctx context.Context, req ctrl.Reque
 		if natsSyncClientStatus.AstraClusterId != "" {
 			log.Info(fmt.Sprintf("Updating CR status, clusterID: '%s'", natsSyncClientStatus.AstraClusterId))
 		}
-		_ = r.updateAstraConnectorStatus(ctx, astraConnector, natsSyncClientStatus)
-		log.Info("assignging to status", "spec", astraConnector.Spec)
+		log.Info("assigning to status", "spec", astraConnector.Spec)
 		astraConnector.Status.ObservedSpec = astraConnector.Spec
-		err = r.Status().Update(ctx, astraConnector)
```

Comment:
Why are you removing this?

Reply:
I removed it because it was redundant since the updateAstraConnectorStatus method does the same update

Reply:
It looks like that update will only be called if the natssync status arg is the same as the status, but we still need to update the observedSpec in other cases.

Reply:
yep, makes sense. I'll add it back in